src/status_im2/common/keychain/events.cljs

```diff
@@ -0,0 +1,110 @@
+(ns status-im2.common.keychain.events
+  (:require [taoensso.timbre :as log]
+            [react-native.platform :as platform]
+            [react-native.keychain :as keychain]
+            [re-frame.core :as re-frame]
+            [utils.re-frame :as rf]
+            [oops.core :as oops]
+            [native-module.core :as native-module]
+            [utils.security.core :as security]))
+
+(defn- check-conditions
+  [callback & checks]
+  (if (= (count checks) 0)
+    (callback true)
+    (let [current-check-fn     (first checks)
+          process-check-result (fn [callback-success callback-fail]
+                                 (fn [current-check-passed?]
+                                   (if current-check-passed?
+                                     (callback-success)
+                                     (callback-fail))))]
+      (current-check-fn (process-check-result
+                         #(apply (partial check-conditions callback) (rest checks))
+                         #(callback false))))))
+
+;; These helpers check if the device is okay to use for password storage
+;; They resolve callback with `true` if the check is passed, with `false` otherwise.
+;; Android only
```

Comment:
nit, why not just change function name and remove comment
```
(defn- android-device-not-rooted?
```

Reply:
good question :) we should ask author of this code :)